include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -245,34 +255,51 @@ class format_embl
         }
         else
         {
-            if (ranges::empty(sequence)) //[[unlikely]]
+            if (std::ranges::empty(sequence)) //[[unlikely]]
                 throw std::runtime_error{"The SEQ field may not be empty when writing embl files."};
 
-            std::ranges::copy(std::string_view{"SQ Sequence "}, stream_it);
-            auto res = std::to_chars(&buffer[0], &buffer[0] + sizeof(buffer), sequence_size);
-            std::copy(&buffer[0], res.ptr, stream_it);
-            std::ranges::copy(std::string_view{" BP;\n"}, stream_it);
-            auto seqChunk = sequence | ranges::view::chunk(60);
-            unsigned int i = 0;
-            size_t bp = 0;
-            for (auto chunk : seqChunk)
+            // write beginning of sequence record
+            stream_it.write_range(std::string_view{"SQ Sequence "});
+            write_number(stream_it, sequence_size);
+            stream_it.write_range(std::string_view{" BP;\n"});
+
+            // write sequence in chunks of 60 bp's with a space after 10 bp's
+
+            auto char_sequence = sequence | views::to_char;
+            size_t written_characters{0};
+
+            auto it = stream_it.write_range(char_sequence | views::take(10)); // write first 10 characters
+            stream_it = ' ';
+
+            while (it != std::ranges::end(char_sequence))
             {
-                std::ranges::copy(chunk | views::to_char
-                                        | ranges::view::chunk(10)
-                                        | views::join(' '), stream_it);
-                ++i;
+                written_characters += 10;
+
+                if (written_characters % 60 == 0)
+                {
+                    write_number(stream_it, written_characters);
+                    detail::write_eol(stream_it, options.add_carriage_return);
+                }
+
+                it = stream_it.write_range(char_sequence | views::drop(written_characters) | views::take(10));
```

Comment:
Is this the same problem as below? Non-random-access ranges are parsed by increasing amount of written_characters ever iteration?

Reply:
I wanted to make a card for embl that the format should be improved for non-random access ranges.. because I thought fasta is the more critical one. I can delete the whole embl transformation altogether if this takes too much time.

Reply:
> I wanted to make a card for embl that the format should be improved for non-random access ranges.. 

That's ok, too of course!